src/utils/helpers/translations/mapColumnsByRole.ts

```diff
@@ -0,0 +1,106 @@
+export enum APPLICATION_COLUMN {
+  LAST_ACTIVE_DATE = 'Last active date',
+  DEADLINE_DATE = 'Deadline date',
+  APPLICATION_NAME = 'Application name',
+  APPLICANT = 'Applicant',
+  COMPANY = 'Company',
+  CONSOLIDATOR = 'Consolidator',
+  STAGE = 'Stage',
+  STATUS = 'Status',
+  ACTIONABLE = 'Actionable',
+}
+
+interface ColumnDetails {
+  headerName: string
+  filters: null | string | Array<string>
+  render?: Function
+}
+
+const allColumns: Array<ColumnDetails> = [
+  {
+    headerName: APPLICATION_COLUMN.LAST_ACTIVE_DATE,
+    filters: 'last-active-date',
+  },
+  {
+    headerName: APPLICATION_COLUMN.DEADLINE_DATE,
+    filters: 'deadline-date',
+  },
+  {
+    headerName: APPLICATION_COLUMN.APPLICATION_NAME,
+    filters: 'search',
+  },
+  {
+    headerName: APPLICATION_COLUMN.APPLICANT,
+    filters: 'applicant',
+  },
+  {
+    headerName: APPLICATION_COLUMN.COMPANY,
+    filters: 'org',
+  },
+  {
+    headerName: APPLICATION_COLUMN.CONSOLIDATOR,
+    filters: 'consolidator',
+  },
+  {
+    headerName: APPLICATION_COLUMN.STAGE,
+    filters: ['search', 'stage'],
+  },
+  {
+    headerName: APPLICATION_COLUMN.STATUS,
+    filters: 'search',
+  },
+  {
+    headerName: '',
+    filters: ['search', 'action'],
+  },
+]
+
+interface ColumnsPerRole {
+  [role: string]: Array<string>
+}
+
+const columnsPerRole: ColumnsPerRole = {
+  applicant: [
+    APPLICATION_COLUMN.LAST_ACTIVE_DATE,
+    APPLICATION_COLUMN.DEADLINE_DATE,
+    APPLICATION_COLUMN.APPLICATION_NAME,
+    APPLICATION_COLUMN.STAGE,
+    APPLICATION_COLUMN.STATUS,
+    APPLICATION_COLUMN.ACTIONABLE,
+  ],
+  reviewer1: [
+    APPLICATION_COLUMN.LAST_ACTIVE_DATE,
+    APPLICATION_COLUMN.APPLICATION_NAME,
+    APPLICATION_COLUMN.APPLICANT,
+    APPLICATION_COLUMN.CONSOLIDATOR,
+    APPLICATION_COLUMN.STAGE,
+    APPLICATION_COLUMN.STATUS,
+  ],
+  reviewer2: [
+    APPLICATION_COLUMN.LAST_ACTIVE_DATE,
+    APPLICATION_COLUMN.APPLICATION_NAME,
+    APPLICATION_COLUMN.APPLICANT,
+    APPLICATION_COLUMN.CONSOLIDATOR,
+    APPLICATION_COLUMN.STAGE,
+    APPLICATION_COLUMN.STATUS,
+  ],
+  supervisor: [
+    APPLICATION_COLUMN.LAST_ACTIVE_DATE,
+    APPLICATION_COLUMN.APPLICATION_NAME,
+    APPLICATION_COLUMN.APPLICANT,
+    APPLICATION_COLUMN.STATUS,
+  ],
+  consolidator: [
+    APPLICATION_COLUMN.LAST_ACTIVE_DATE,
+    APPLICATION_COLUMN.APPLICATION_NAME,
+    APPLICATION_COLUMN.APPLICANT,
+    APPLICATION_COLUMN.STATUS,
+  ],
+}
+
+export default (userRole: string): Array<ColumnDetails> => {
+  const columns: Array<ColumnDetails> = allColumns.filter(({ headerName }) =>
+    columnsPerRole[userRole] ? columnsPerRole[userRole].includes(headerName) : false
```

Comment:
I think a much more efficient mapping would be:
```
const columns: Array<ColumnDetails> = columnsPerRole[userRole].map((key) => allColumns[key] )
```

And then instead of `allColumns` being an array, it should be an object with the unique name as the key, i.e.:
```
const allColumns = {
  LAST_ACTIVE_DATE: {
    headerName: APPLICATION_COLUMN.LAST_ACTIVE_DATE,
    filters: 'last-active-date',
  },
  DEADLINE_DATE: {
    headerName: APPLICATION_COLUMN.DEADLINE_DATE,
    filters: 'deadline-date',
  },
  APPLICATION_NAME: {
    headerName: APPLICATION_COLUMN.APPLICATION_NAME,
    filters: 'search',
  },
  APPLICANT: {
    headerName: APPLICATION_COLUMN.APPLICANT,
    filters: 'applicant',
  },
  COMPANY: {
    headerName: APPLICATION_COLUMN.COMPANY,
    filters: 'org',
  },
  CONSOLIDATOR: {
    headerName: APPLICATION_COLUMN.CONSOLIDATOR,
    filters: 'consolidator',
  },
  STAGE: {
    headerName: APPLICATION_COLUMN.STAGE,
    filters: ['search', 'stage'],
  },
  STATUS: {
    headerName: APPLICATION_COLUMN.STATUS,
    filters: 'search',
  },
  ACTIONABLE: {
    headerName: '',
    filters: ['search', 'action'],
  },
}
```

And then the columnsPerRow would just list the keys for each role type, e.g.:
```
applicant: [
    LAST_ACTIVE_DATE,
    DEADLINE_DATE,
    APPLICATION_NAME,
    STAGE,
    STATUS,
    ACTIONABLE,
  ]
```

It'd still return an array of column objects, but it's a linear operation to build it.

Reply:
Yea, that doesn't work exactly like that. I'm still trying to figure out how to create a type to list the possible columns, which is how I think this could work, just saying each constant of a user role is an array of that type. But I still need a little more research to make this one work.

Reply:
Ok. I made quite a big change now. I'm not so happy with the way we can have types using enums, but I guess that's how things are. Please have another look and see what do you think. Mind that there is a new folder called `model` where I tried to keep this constants that are being used as types, but aren't types. So I guess models?